packages/kbn-i18n/src/react/provider.tsx

```diff
@@ -29,12 +29,26 @@ import * as i18n from '../core';
  * IntlProvider should wrap react app's root component (inside each react render method).
  */
 export class I18nProvider extends React.PureComponent {
-  public static propTypes = {
-    children: PropTypes.object,
-  };
+  public static propTypes = { children: PropTypes.element.isRequired };
+  public static contextTypes = { intl: intlShape };
+  public static childContextTypes = { intl: intlShape };
+
+  public getChildContext() {
+    if (this.context.intl && i18n.isPseudoLocale()) {
```

Comment:
__nit:__ please add comment for the future readers why and what we're doing here.